src/palimpzest/elements/groupbysig.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+from typing import Any, Dict
+from palimpzest.elements import Field, OperatorDerivedSchema
+
+#signature for a group by aggregate that applies
+# group and aggregation to an input tuple
+class GroupBySig:
```

Comment:
Should GroupBySig take as input the Schema that it will be applied to?  If it did, then we could test for syntactic compatibility at the moment the GroupBySig is defined.  E.g., if you try to group by a field that doesn't exist in the target schema, then you can't create the GroupBySig.

Is it useful to have GroupBySigs that exist independently of the Schema that is being grouped-by?  I am not sure about that.

I am honestly not sure what's best here.

Reply:
Yes this is probably useful to check here -- I was sort of in the mindset of a query planner where this kind of semantic compatibility would be handled by the layer above but since this is the interface programmers directly write having some decent error checking is a good idea.   I can add that.